pkg/operator/nodekubeconfigcontroller/nodekubeconfigcontroller.go

```diff
@@ -147,6 +148,10 @@ func ensureNodeKubeconfigs(ctx context.Context, client coreclientv1.CoreV1Interf
 		requiredSecret.StringData[k] = data
 	}
 
+	requiredSecret.Annotations = map[string]string{
```

Comment:
this will completely wipe out already set annotations. change to:
1. first check if the map is nil
2. if nil then allocate memory
3. add the new annotation.

Reply:
I expect `ApplySecret` to preserve the existing ones. But its a good idea not to rely on that, I agree

Reply:
If I add an annotation to the manifest stored in the assets this code will remove it so we have to fix it.